### PR TITLE
Adding a delete functionality for global/static info_dict

### PR DESCRIPTION
**Pull request by raghavsake**

### CODE REVIEW

File: shell_util/remote_connection.py [hunk 1]
```diff
@@ -45,3 +45,10 @@ def __new__(cls, *args, **kwargs):
         obj.ssh_connect_with_retries(server.ip, server.ssh_username,
                                      server.ssh_password, server.ssh_key)
         return obj
+
+    @staticmethod
+    def delete_info_for_server(server, ipaddr=None):
+        ipaddr = ipaddr if ipaddr else server.ip
```
Comment (ashwin2002) [inline shell_util/remote_connection.py]: This can be simplified as,
   ipaddr = ipaddr or server.ip

File: shell_util/remote_connection.py [hunk 2]
```diff
@@ -45,3 +45,10 @@ def __new__(cls, *args, **kwargs):
...
+        if ipaddr in RemoteMachineShellConnection.__info_dict:
+            del RemoteMachineShellConnection.__info_dict[ipaddr]
```
Comment (ashwin2002) [inline shell_util/remote_connection.py:53]: Should we also close the connection here ?

File: shell_util/remote_connection.py (hunk 1 above)
Comment (raghavsake) [reply shell_util/remote_connection.py]: I have made the changes in latest commit

File: shell_util/remote_connection.py (hunk 2 above)
Comment (raghavsake) [reply shell_util/remote_connection.py:53]: We don't store the connection in this particular class. All we are storing is the info dictionary to fetch the right class.

**Review by ashwin2002 — APPROVED** [review]
ACK

**Pull request merged**